onchain/wallet/wallet.go

```diff
@@ -109,6 +112,14 @@ type Config struct {
 
 var config *Config
 
+type syncListener struct {
+	accounts []uint64
+	done     chan bool
+}
+
+var syncListeners []*syncListener
+var syncListenersLock = sync.Mutex{}
```

Comment:
These should be declared as fields of `Wallet`

Reply:
No, they have to be global, since the gdk notification unfortunately is global aswell and doesn't contain information about the wallet.